nilearn/maskers/surface_masker.py

```diff
@@ -182,10 +182,13 @@ def _fit_mask_img(self, img):
         img = at_least_2d(img)
         mask_data = {}
         for part, v in img.data.parts.items():
-            mask_data[part] = v.astype("float32")
-            non_finite_mask = np.logical_not(np.isfinite(mask_data[part]))
-            mask_data[part][non_finite_mask] = 0
-            mask_data[part] = mask_data[part].astype("bool").all(axis=1)
+            # mask out vertices with NaN or infinite values
+            mask_data[part] = np.isfinite(v.astype("float32")).all(axis=1)
+            warn(
+                "Non-finite values detected in the input image. "
+                "The computed mask will mask out these vertices.",
+                stacklevel=find_stack_level(),
+            )
```

Comment:
This line:

```
mask_data[part] = np.isfinite(v.astype("float32")).all(axis=1)
```

does not guarantee that there are infinite values. Shouldn't there be a condition that checks if infinite values exist and then give the warning? May be I am missing something.

Reply:
yeah I think that otherwise the warning is always thrown, no?

Reply:
- BTW #5359 should include a single function to help dealing with non finite values in several places

Reply:
oops yes you're right